code/control/DNRoot.php

```diff
@@ -918,11 +963,13 @@ public function doCreateEnvironment($data, Form $form) {
 			return new SS_HTTPResponse('Not allowed to create environments for this project', 401);
 		}
 
+		// Set the environment type so we know what we're creating.
+		$data['EnvironmentType'] = $project->AllowedEnvironmentType;
+
 		$job = DNCreateEnvironment::create();
 		$environment = DNEnvironment::create();
 		$environment->ProjectID = $project->ID;
-		$environment->Usage = ''; // todo fill out proper usage here
-		$environment->Name = $project->Cluster . '-' . $project->Name . '-' . $data['Usage'] . '-';
+		$environment->Usage = $datap['Usage'];
 		$environment->write();
 
 		$job->Data = serialize($data);
```

Comment:
We'll need to validate that form input when we roll it out to users... otherwise people can switch roles and profiles... Maybe for now put a big TODO here? :-)